src/tools/tests/kotlin-class-generator-test.ts

```diff
@@ -119,58 +119,173 @@ describe('kotlin-class-generator', () => {
         
         override var entityId = ""`
   ));
+  it('generates empty schema', async () => await assertSchemas(
+    `particle T
+         h1: reads {}`,
+    [`Schema.EMPTY`]
+  ));
+  it('generates a schema with multiple names', async () => await assertSchemas(
+    `particle T
+         h1: reads Person Friend Parent {}`, [
+`Schema(
+    setOf(SchemaName("Person"), SchemaName("Friend"), SchemaName("Parent")),
+    SchemaFields(
+        singletons = emptyMap(),
+        collections = emptyMap()
+    ),
+    "cd956ff7d2d4a14f434aa1427b84097d5c47037b",
+    refinement = { _ -> true },
+    query = null
+)`
+    ]
+  ));
+  it('generates a schema with primitive fields', async () => await assertSchemas(
+    `particle T
+         h1: reads Person {name: Text, age: Number, friendNames: [Text]}`, [
+`Schema(
+    setOf(SchemaName("Person")),
+    SchemaFields(
+        singletons = mapOf("name" to FieldType.Text, "age" to FieldType.Number),
+        collections = mapOf("friendNames" to FieldType.Text)
+    ),
+    "accd28212161fc896d658e8c22c06051d1239e18",
+    refinement = { _ -> true },
+    query = null
+)`
+    ]
+  ));
+  it('generates schemas for a reference', async () => await assertSchemas(
```

Comment:
You could add a test to see if it generates multiple schemas for tuple handles. I'm not sure if this is needed, I like the test cases you have here.

Reply:
Done!